drupal_seamless_cilogon.module

```diff
@@ -0,0 +1,166 @@
+  );
+
+  $cookie_value = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_value', null);
+  if (strlen($cookie_value) == 0) {
+    $cookie_value =  "INITIAL_DOMAIN=$site_name";
+  }
+
+  $cookie_expiration = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_expiration', null);
+  $cookie_expiration = strlen($cookie_expiration) == 0
+    ? strtotime('+18 hours')  // default to +18 hours
+    : strtotime($cookie_expiration);  // use value from form
+  
+  // Set the cookie domain (unless debugging)
+  $cookie_domain = $seamless_debug ? null : 'access-ci.org';
+
+  // set the cookie. 
+  setcookie(
+    $cookie_name,
+    $cookie_value,
+    $cookie_expiration,
+    '/', // path
+    $cookie_domain, // domain
+  );
+
+  if ($seamless_debug) {
+    $msg =  __FUNCTION__ . "() - set cookie, name = $cookie_name, value = $cookie_value, expiration = $cookie_expiration "
+      . " = " . date("Y-m-d H:i:s", $cookie_expiration) . ", domain = $cookie_domain"
+      . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+    \Drupal::messenger()->addStatus($msg);
+    \Drupal::logger('seamless_cilogon')->debug($msg);
+  }
+}
+
+/**
+ * Implements hook_user_logout().  When logging out from ACCESS Support, remove the cookie and 
+ * redirect to https://cilogon.org/logout/.
+ */
+function drupal_seamless_cilogon_user_logout($account)
+{
+  if (!verify_domain()) {
+    return;
+  }
+  
+  $cookie_name = \Drupal::state()->get(
+    'drupal_seamless_cilogon.seamless_cookie_name',
+    DrupalSeamlessCilogonEventSubscriber::seamlessCookieName
+  );
+
+  // Set cookie in the past and then remove it.
+  setcookie(
+    $cookie_name,
+    '',
+    strtotime('-1 hour'),
+    '/'
+  );
+  unset($_COOKIE[$cookie_name]);
+
+
+  $seamless_debug = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_debug', true);
+  if ($seamless_debug) {
+    $msg =  __FUNCTION__ . "() - unset cookie, name = $cookie_name"
+      . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+    \Drupal::messenger()->addStatus($msg);
+    \Drupal::logger('seamless_cilogon')->debug($msg);
+  }
+}
+
+
+/**
+ * The ACCESS support portal uses the domain access module.  If this module
+ * is in use, we only want to set cookies for the 'access-support' 
+ * module.  
+ * 
+ * This function checks if the domain access module is in use, and 
+ * if so, returns FALSE if the current domain name is not 'access-support'.
+ * 
+ * Otherwise it returns true.
+ * 
+ * @return bool whether to proceed with the cookie logic in invoking code
+ * 
+ */
+function verify_domain()
+{
+  // Verify the domain module is installed.  
+  $moduleHandler = \Drupal::service('module_handler');
+  if (!$moduleHandler->moduleExists('domain')) {
+    return false;
+  }
+
+  // Get the current domain.
+  // If the domain is null, simply return true.  we only want to check that the 
+  // current domain is 'access-support' if the domain access module is installed.
```

Comment:
I need some help with verifying the null return for no domain access module.  Not sure how to do a new drupal install.

Reply:
code removed!